crates/ra_hir/src/ty.rs

```diff
@@ -216,6 +216,9 @@ pub enum Ty {
     /// A tuple type.  For example, `(i32, bool)`.
     Tuple(Arc<[Ty]>),
 
+    /// A array type.  For example, `[i32]`.
```

Comment:
we already have `Slice` above. The difference between `Array` and `Slice` would be that `Array` has a known length (which we can leave out for now since we don't care about it), but then your example is wrong ;) Also, an array should only have one inner type -- all elements of the array need to have the same type.

Also can you move this up to be next to `Slice`?